core/src/services/supabase/core.rs

```diff
@@ -0,0 +1,290 @@
+    pub bucket: String,
+    pub endpoint: String,
+
+    /// The key used for authorization, initialized by environment variable you designated.
+    /// Normally it is rather an anon_key(Client key) or an service_role_key(Secret Key)
+    pub auth_key: Option<HeaderValue>,
+    /// This is true if the service_role_key is loaded, false if the anon_key is loaded
+    pub auth: bool,
+
+    pub http_client: HttpClient,
+}
+
+impl Debug for SupabaseCore {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("SupabaseCore")
+            .field("root", &self.root)
+            .field("bucket", &self.bucket)
+            .field("endpoint", &self.endpoint)
+            .finish_non_exhaustive()
+    }
+}
+
+impl SupabaseCore {
+    pub fn new(root: &str, bucket: &str, endpoint: &str, client: HttpClient) -> Self {
+        Self {
+            root: root.to_string(),
+            bucket: bucket.to_string(),
+            endpoint: endpoint.to_string(),
+            auth_key: None,
+            auth: false,
+            http_client: client,
+        }
+    }
+
+    /// load auth key tries to load the authorization key from the environment variable
+    /// - if the key is set by user using Builder, this return directly
+    /// - if the key is not set by user, this will try to load the service key first, then the anon key
+    pub fn load_auth_key(&mut self) {
+        // if set by user, return directly
+        if self.auth_key.is_some() {
+            return;
+        }
+
+        // load secret key first
+        if let Ok(v) = std::env::var(constants::SUPABASE_AUTH_SERVICE_KEY) {
+            self.auth_key = Some(HeaderValue::from_str(&v).unwrap());
+            self.auth = true;
+            return;
+        }
+
+        // if not loaded, load anon key then
+        if let Ok(v) = std::env::var(constants::SUPABASE_AUTH_ANON_KEY) {
+            self.auth_key = Some(HeaderValue::from_str(&v).unwrap());
+            self.auth = false;
+            return;
+        }
+
+        // the key should always be loaded
+        unreachable!(
+            "The authorization key is not set, you may set it in your builder or through environment variable {} or {}",
+            constants::SUPABASE_AUTH_ANON_KEY,
+            constants::SUPABASE_AUTH_SERVICE_KEY,
+        )
+    }
+
+    pub fn sign<T>(&self, req: &mut Request<T>) -> Result<()> {
+        if let Some(k) = &self.auth_key {
+            let v = HeaderValue::from_str(&format!("Bearer {}", k.to_str().unwrap())).unwrap();
+            req.headers_mut().insert(constants::AUTH_HTTP_HEADER_KEY, v);
+            Ok(())
+        } else {
+            Err(new_request_sign_error(anyhow!(
+                "The anon key is not loaded"
+            )))
+        }
+    }
+}
+
+// requests
+impl SupabaseCore {
+    // ?: this defaults the bucket id to be the bucket name
+    pub fn supabase_create_bucket_request(&self, public: bool) -> Result<Request<AsyncBody>> {
```

Comment:
No, we don't need to support create_bucket. Bukcet must be created before starting use opendal.